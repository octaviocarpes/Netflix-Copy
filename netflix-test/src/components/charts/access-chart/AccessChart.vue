<template>
  <div class="access-chart-container">
    <canvas id="access-chart"></canvas>
    <div class="access-data">
      <h2>Logins:</h2><h3>{{ logins }}</h3>
    </div>
    <div class="access-data">
      <h2>Registers:</h2><h3>{{ registers }}</h3>
    </div>
  </div>
</template>

<script>
import Chart from "chart.js";
import { ACCESS_METRICS } from "@/controllers/access.metrics.controller";

export default {
  name: "AccessChart",

  mounted() {
    this.initChart();
  },

  methods: {
    initChart() {
      const ctx = document.getElementById("access-chart").getContext("2d");
      const chart = new Chart(ctx, {
        type: "pie",
        data: {
          labels: ["Logins", "Registers"],
          datasets: [
            {
              label: "Access Data",
              backgroundColor: this.generateColors(),
              borderColor: "rgb(255, 255, 255)",
              data: [ACCESS_METRICS.getLogins(), ACCESS_METRICS.getRegisters()],
            }
          ]
        }
      });
    },

    generateColors() {
      return [this.random_rgba(), this.random_rgba()]
    },

    random_rgba() {
      let o = Math.round,
        r = Math.random,
        s = 255;
      return `rgba(${o(r() * s)}, ${o(r() * s)}, ${o(r() * s)}, 1)`;
    }
  },

  computed: {
    logins() {
      return ACCESS_METRICS.getLogins();
    },

    registers() {
      return ACCESS_METRICS.getRegisters();
    }
  }
}
</script>

<style lang="scss" scoped>
@import "./style";
</style>
