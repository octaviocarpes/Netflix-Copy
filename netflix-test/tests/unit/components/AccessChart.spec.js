import Vue from "vue";
import AccessChart from "@/components/charts/access-chart/AccessChart.vue";

describe("AccessChart.spec.js", () => {
  let cmp, vm;

  beforeEach(() => {
    cmp = Vue.extend(AccessChart);
    vm = new cmp({}).$mount();
  });

  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});