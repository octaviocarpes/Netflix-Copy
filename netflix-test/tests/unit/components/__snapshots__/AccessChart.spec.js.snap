// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessChart.spec.js has the expected html structure 1`] = `
<div
  class="access-chart-container"
>
  <canvas
    id="access-chart"
  />
   
  <div
    class="access-data"
  >
    <h2>
      Logins:
    </h2>
     
    <h3>
      0
    </h3>
  </div>
   
  <div
    class="access-data"
  >
    <h2>
      Registers:
    </h2>
     
    <h3>
      0
    </h3>
  </div>
</div>
`;
